### ISSUE TITLE
Include a Validation loss in the hyperparameter_tuning() function

### ISSUE THREAD

**Issue by BoboyeOkeya** [BoboyeOkeya/Physics_Informed_Neural_Network#2]
It seems I should not have used a training loss as the decision factor for selecting the best performing combination of hyperparameters.

I should have created a validation data (with a smaller size,about 10-20% of training data size, for computational efficiency) and use the validation loss as the decision factor.

Hence, I should possibly follow the steps below:
a. For each set of hyperparameters:
-Train models using the current hyperparameters on the training data.
-Evaluate the models on the validation data to compute the validation loss or performance metric.
b. If validation loss is the lowest, then select set as optimal
